### PR TITLE
HCL2\Terraform v0.12 conversion

### PR DESCRIPTION
**Pull request by twistedgrim**
##### Corresponding Issue(s):
https://jira.rax.io/browse/MPCSUPENG-871

##### Summary of change(s):
- Module updated from HCL1 syntax to HCL2
- Module configured with a minimum terraform version of 0.12.0 and minimum AWS provider version of 2.2.
- Formatting and code changes due to HCL2 conversion and to meet Rackspace Terraform Style Guide requirements

##### Reason for Change(s):
Terraform 0.12.0 Support

##### Will the change trigger resource destruction or replacement? If yes, please provide justification:
No

##### Does this update/change involve issues with other external modules? If so, please describe the scenario.
No, none of the changes should result in the replacement of resources, however several resources may be updated during the process, such as tags, SSM Document spacing, etc.

##### If input variables or output variables have changed or has been added, have you updated the README?
Readme should be up to date.

##### Do examples need to be updated based on changes?
Examples were converted during the process and updated to the expected release number.

##### Note to the PR requester about Closing PR's
Please message the person that opened the issue when auto closing it on slack, as well as any other stake holders of deep interest. Only close the issue if you believe that the issue is fully resolved with this PR.

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -1,53 +1,65 @@
...
+  description = <<EOF
+Create Route 53 internal zone record for the resource Default is \"false\".
+EOF
 
-variable "environment" {
-  description = "Application environment for which this network is being created. e.g. Development/Production"
-  type        = "string"
-  default     = "Development"
+  type    = string
+  default = "false"
```
Comment (johnctitus) [inline variables.tf]: I would suggest updating this variable, and the `create_parameter_store_entries` . and `encrypted` variables all to bool types and remove the quotes around the default values.

File: variables.tf [hunk 2]
```diff
@@ -1,53 +1,65 @@
...
   default     = {}
 }
 
-##################
-# EFS Core Options
-##################
+variable "cw_burst_credit_period" {
+  description = "The number of periods over which the EFS Burst Credit level is compared to the specified threshold."
+  type        = string
+  default     = "12"
```
Comment (johnctitus) [inline variables.tf]: I would recommend changing the type of this variable and the `cw_burst_credit_threshold` variable to number, and removing the quotes around the default values.

File: README.md [hunk 3]
```diff
@@ -16,34 +16,52 @@ module "efs" {
...
+
+\*Several changes were required while adding terraform 0.12 compatibility.  The following changes should be
+\*made when upgrading from a previous release to version 0.12.0 or higher.
+
+\*### Module variables
+
+\*The following module variables were updated to better meet current Rackspace style guides:
+
+\*- `custom_tags` -> `tags`
+\*- `create_internal_dns_record` -> 'create\_internal\_zone\_record'
```
[the remark this answers is not in the document]
Comment (twistedgrim) [reply README.md]: all fixed

File: main.tf [hunk 4]
```diff
@@ -20,81 +20,113 @@
...
+ *Several changes were required while adding terraform 0.12 compatibility.  The following changes should be
+ *made when upgrading from a previous release to version 0.12.0 or higher.
+ *
+ *### Module variables
+ *
+ *The following module variables were updated to better meet current Rackspace style guides:
+ *
+ *- `custom_tags` -> `tags`
+ *- `create_internal_dns_record` -> 'create_internal_zone_record'
+ *
```
[the remark this answers is not in the document]
Comment (twistedgrim) [reply main.tf]: got it

File: variables.tf (hunk 1 above)
Comment (twistedgrim) [reply variables.tf]: I had done some of them but got messed up when I reordered. Its all fixed now.

File: variables.tf (hunk 2 above)
Comment (twistedgrim) [reply variables.tf]: all good now

**Pull request merged**